From 86399b9825386337bd690bcb8a3c065358799e98 Mon Sep 17 00:00:00 2001
From: Jens Axboe <axboe@fb.com>
Date: Tue, 19 Apr 2016 13:07:22 -0400
Subject: blk-wb: replace multiplier by shift

Instead of multiplying by 10 and dividing by the square root of 100,
shift up by 4 and divide by the square root of a shift by 8. This
gives us the same result/accuracy, but eliminates the need for a
multiplication.

Now we just have to get rid of the division...

Signed-off-by: Jens Axboe <axboe@fb.com>
---
 block/blk-wb.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/block/blk-wb.c b/block/blk-wb.c
index 30c86a5..859cbb2 100644
--- a/block/blk-wb.c
+++ b/block/blk-wb.c
@@ -290,8 +290,8 @@ static void rwb_arm_timer(struct rq_wb *rwb)
 	 * inverse square root calculation. Since we only do this for
 	 * every window expiration, it's not a huge deal, though.
 	 */
-	rwb->cur_win_nsec = div_u64(rwb->win_nsec * 10,
-					int_sqrt((rwb->scale_step + 1) * 100));
+	rwb->cur_win_nsec = div_u64(rwb->win_nsec << 4,
+					int_sqrt((rwb->scale_step + 1) << 8));
 	expires = jiffies + nsecs_to_jiffies(rwb->cur_win_nsec);
 	mod_timer(&rwb->window_timer, expires);
 }
-- 
cgit v0.11.2

